### PR TITLE
add V1_getExtraExecutionInputElements

### PR DESCRIPTION
**Pull request by MauricioUyaguari**
Add V1_getExtraExecutionInputElements to send additional elements as part of query execution call (#219 ).

### CODE REVIEW

File: packages/legend-studio/src/models/protocols/pure/v1/V1_PureGraphManager.ts [hunk 1]
```diff
@@ -1794,6 +1794,9 @@ export class V1_PureGraphManager extends AbstractPureGraphManager {
     const graphData = this.getFullGraphModelData(graph);
     /* @MARKER: NEW ELEMENT TYPE SUPPORT --- consider adding new element type handler here whenever support for a new element type is added to the app */
     const prunedGraphData = new V1_PureModelContextData();
+    const extraExecutionClasses = this.pureProtocolProcessorPlugins.flatMap(
+      (e) => e.V1_getExtraMappingExecutionClasses?.() ?? [],
```
Comment (MauricioUyaguari) [inline packages/legend-studio/src/models/protocols/pure/v1/V1_PureGraphManager.ts:1798]: to rename `getExtraExuectionInputElements`

File: .changeset/fast-fireants-fix.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+'@finos/legend-studio': patch
+'@finos/legend-studio-plugin-query-builder': patch
+---
+
+Add V1_getExtraMappingExecutionClasses to send extra plugin elements as part of query execution call.
```
Comment (MauricioUyaguari) [inline .changeset/fast-fireants-fix.md:6]: `V1_getExtraMappingExecutionClasses`

File: packages/legend-studio/src/models/protocols/pure/PureProtocolProcessorPlugin.ts [hunk 3]
```diff
@@ -53,4 +53,6 @@ export abstract class PureProtocolProcessorPlugin extends AbstractPlugin {
   V1_getExtraElementTransformers?(): V1_ElementTransformer[];
 
   V1_getExtraSourceInformationKeys?(): string[];
+
+  V1_getExtraMappingExecutionClasses?(): Clazz<V1_PackageableElement>[];
```
Comment (akphi) [inline packages/legend-studio/src/models/protocols/pure/PureProtocolProcessorPlugin.ts:57]: let's change this to `V1_getExtraExuectionInputElements?(): V1_PackageableElement[]` and add a little bit of doc to this

File: .changeset/fast-fireants-fix.md (hunk 2 above)
Comment (akphi) [reply .changeset/fast-fireants-fix.md:6]: Yeah, let's give a bit of background here on why we need this.

**Review by akphi — APPROVED** [review]

**Pull request merged**